crates/services/block_aggregator_api/src/tests.rs

```diff
@@ -209,3 +213,70 @@ async fn run__get_current_height__returns_expected_height() {
     // cleanup
     drop(_block_sender);
 }
+
+#[tokio::test]
+async fn run__new_block_subscription__sends_new_block() {
+    let mut rng = StdRng::seed_from_u64(42);
+    // given
+    let (api, sender) = FakeApi::new();
+    let db = FakeDB::new();
+    let (source, source_sender) = FakeBlockSource::new();
+    let mut srv = BlockAggregator::new(api, db, source);
+
+    let expected_block = Block::random(&mut rng);
+    let expected_height = BlockHeight::from(123u32);
+    let mut watcher = StateWatcher::started();
+    let (query, mut response) = BlockAggregatorQuery::new_block_subscription();
+
+    // when
+    sender.send(query).await.unwrap();
+    let _ = srv.run(&mut watcher).await;
+    let event = BlockSourceEvent::NewBlock(expected_height, expected_block.clone());
+    source_sender.send(event).await.unwrap();
+    let _ = srv.run(&mut watcher).await;
+
+    // then
+    tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
+    let (actual_height, actual_block) =
+        tokio::time::timeout(tokio::time::Duration::from_secs(1), response.recv())
+            .await
+            .unwrap()
+            .unwrap();
```

Comment:
instead of the sleep, we can turn this into a loop

```suggestion
    let (actual_height, actual_block) = tokio::time::timeout(
    tokio::time::Duration::from_secs(1),
    async {
        loop {
            if let Ok(result) = response.try_recv() {
                return result;
            }
            tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
        }
    }
).await.unwrap().unwrap();
```